arc4.c

```diff
@@ -200,8 +204,8 @@ arc4_ARC4_crypt(struct arc4_ARC4 *self, PyObject *arg)
     outputBytes = PyBytes_FromStringAndSize(NULL, size);
     output = PyBytes_AS_STRING(outputBytes);
     Py_BEGIN_ALLOW_THREADS
-        arc4_crypt(&(self->state), (const unsigned char *)input,
-                   (unsigned char *)output, size);
+        arc4_crypt(&(self->state), (const uint_fast8_t *)input,
+                   (uint_fast8_t *)output, size);
```

Comment:
Casts like `(uint_fast8_t *)` seems unsafe in a system that defines `sizeof(uint_fast8_t)` not 1.
To avoid this pitfall, it should allocate memory with size `size * sizeof(uint_fast8_t)` although it makes this function much slower.